pkg/R/DataFrame.R

```diff
@@ -78,15 +75,96 @@ setMethod("printSchema",
 #' df <- jsonFile(sqlCtx, path)
 #' dfSchema <- schema(df)
 #'}
-
 setGeneric("schema", function(x) { standardGeneric("schema") })
 
+#' @rdname schema
+#' @export
 setMethod("schema",
           signature(x = "DataFrame"),
           function(x) {
             structType(callJMethod(x@sdf, "schema"))
           })
 
+#' Explain
+#' 
+#' Print the logical and physical Catalyst plans to the console for debugging.
+#' 
+#' @param x A SparkSQL DataFrame
+#' @param extended Logical. If extended is False, explain() only prints the physical plan.
+#' @rdname explain
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' explain(df, TRUE)
+#'}
+setGeneric("explain", function(x, ...) { standardGeneric("explain") })
+
+#' @rdname explain
+#' @export
+setMethod("explain",
+          signature(x = "DataFrame"),
+          function(x, extended = FALSE) {
+            callJMethod(x@sdf, "explain", extended)
+          })
+
+#' isLocal
+#'
+#' Returns True if the `collect` and `take` methods can be run locally
+#' (without any Spark executors).
+#'
+#' @param x A SparkSQL DataFrame
+#'
+#' @rdname isLocal
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' isLocal(df)
+#'}
+setGeneric("isLocal", function(x) { standardGeneric("isLocal") })
+
+#' @rdname isLocal
+#' @export
+setMethod("isLocal",
+          signature(x = "DataFrame"),
+          function(x) {
+            callJMethod(x@sdf, "isLocal")
+          })
+
+#' ShowDF
+#'
+#' Print the first numRows rows of a DataFrame
+#'
+#' @param x A SparkSQL DataFrame
+#' @param numRows The number of rows to print. Defaults to 20.
+#'
+#' @rdname showDF
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' showDF(df)
+#'}
+setGeneric("showDF", function(x,...) { standardGeneric("showDF") })
```

Comment:
I tried that actually, but scratched it because the generic is fixed to one argument, so we wouldn't be able to add numRows as an additional argument in the overloaded method.